WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -366,6 +366,11 @@ public async Task<OutputRegistrationResponse> RegisterOutputAsync(OutputRegistra
 
 				var credentialAmount = -request.AmountCredentialRequests.Delta;
 
+				if (StandardScripts.GetTemplateFromScriptPubKey(request.Script) == null)
+				{
+					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.NonStandardOutput, $"Round ({request.RoundId}): Non standard output.");
+				}
+
 				if (!request.Script.IsScriptType(ScriptType.P2WPKH))
```

Comment:
Why is this restriction?

Reply:
It was there before, and it make more sense to generalize and parameterized at a later point than remove outright. Non-standard scripts should never be allowed, but on top of that restriction we might want to limit specific rounds to specific script types based on round parameters for specific rounds.